### PR TITLE
Feat/enhance supplier dashboard

### LINKED ISSUE #165: feat: enhance supplier Dashboard
The Supplier Dashboard Filtering mechanism should be enhanced. One should be able to:
- Select a customer
- Select a product the customer buys
- Select location (BPNS and bpna)

Data should be filtered accordingly:
- currently we only have stocks as real data. Mock the other data.
- Stocks should be shown in both views (Customer = Reported Product Stock, Your stock = Product Stock allocated to the selected partner for given material at given location)

One should be able to trigger a stock update for the customer's stock one can see.

### PR DESCRIPTION
**Pull request by PaMarzec**
## Description
closes #165 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: frontend/src/views/SupplierDashboard.vue [hunk 1]
```diff
@@ -24,23 +24,39 @@
...
                     </select>
+                    <div class="mt-2">
+                        <select v-model="dropdownBPNA" id="dropdown-bpna" name="ddbpna" :disabled="dropdownBPNS === ''" class="w-60 py-2 px-4 bg-gray-200 text-gray-700 border border-gray-200 rounded focus:bg-white focus:outline-none focus:border-gray-500">
+                            <option disabled value="" selected hidden>Choose an address</option>
+                            <option value="allAddresses" @click="emptyTotalDemandArray();aggregateAllAddressesData()">All addresses</option>
+                            <option v-for="(item,index) in dropdownBPNS.addresses" :value="item" @click="emptyTotalDemandArray();mockDataInput(index)"> {{item.streetAndNumber + ", " + item.zipCodeAndCity}} </option>
+                        </select>
+                    </div>
                 </div>
             </div>
```
Comment (tom-rm-meyer-ISST) [inline frontend/src/views/SupplierDashboard.vue:62]: Updating data from customer is missing for Stocks, when triggering the button

File: frontend/src/views/SupplierDashboard.vue [hunk 2]
```diff
@@ -129,122 +157,235 @@
...
+                    var value = parseFloat(filteredStocksMap.get(uom)) + parseFloat(this.fetchedCustomerStocks[i].quantity)
+                    filteredStocksMap.set(uom,value)
+                    // If no then just add (key,value)
+                } else {
+                    filteredStocksMap.set(uom, this.fetchedCustomerStocks[i].quantity)
+                }
+            }
+        }
+        for(let [key,value] of filteredStocksMap.entries()){
+            string += value + key + ", "
```
Comment (tom-rm-meyer-ISST) [inline frontend/src/views/SupplierDashboard.vue:320]: please add blank and don't add a comma if only one entry is given.

File: frontend/src/views/SupplierDashboard.vue [hunk 3]
```diff
@@ -129,122 +157,235 @@
...
+        endpointCustomers: import.meta.env.VITE_ENDPOINT_CUSTOMER,
+        endpointAllProductStocks: import.meta.env.VITE_ENDPOINT_PRODUCT_STOCKS,
+        endpointReportedProductStocks: import.meta.env.VITE_ENDPOINT_REPORTED_PRODUCT_STOCKS,
         // Customer dropdown
         dropdownCustomer: "",
         // Material dropdown
-        dropdownMaterial: "",
+        dropdownProducts: "",
+        // BPNS dropdown
+        dropdownBPNS: "",
```
Comment (tom-rm-meyer-ISST) [inline frontend/src/views/SupplierDashboard.vue:180]: Please bind to customer's sites and addresses.

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Overall great effort. 
Beside a few major things please:
- bind sites and addresses from partner
- fix logic of aggregation with partner (nothing happens if bpna is set to allAddresses)
- update partner data (in meantime also update of reported product stocks have been implemented)

File: frontend/src/views/SupplierDashboard.vue [hunk 4]
```diff
@@ -129,122 +157,235 @@
...
+            .then((res) => res.json())
+            .then((data) => (this.fetchedCustomerStocks = data))
+            .catch((err) => console.log(err));
+    },
+    filterCustomerStocks: function (bpnl,bpns,bpna){
+        let filteredStocksMap = new Map();
+        let string = "";
+        for(let i=0;i <this.fetchedCustomerStocks.length;i++){
+            // Filter through this.fetchedCustomerStocks where bpnl,bpns and bpna are equal to the chosen dropdowns
+            if (bpnl === this.fetchedCustomerStocks[i].partner.bpnl && bpns === this.fetchedCustomerStocks[i].stockLocationBpns && bpna === this.fetchedCustomerStocks[i].stockLocationBpna) {
```
[the remark this answers is not in the document]
Comment (PaMarzec) [reply frontend/src/views/SupplierDashboard.vue:305]: all addresses  is now included and material is already filtered in getCustomerStocks()

File: frontend/src/views/SupplierDashboard.vue (hunk 3 above)
Comment (PaMarzec) [reply frontend/src/views/SupplierDashboard.vue:180]: The sites are already bound as an object into the dropdownBPNS variable. Same applies to the addresses and the dropdownBPNA variable.

**Review by tom-rm-meyer-ISST — APPROVED** [review]
LGTM, thanks for contributing!

**Pull request merged**